### PR TITLE
Update dhm.h

### PR DESCRIPTION
**Pull request by RosZad**
Minor documentation improvements:
*Standardized file brief description.
*Separated return statements.
*Reordered tags within documentation blocks so that params and returns are last in block.
*p_rng descriptions changed from "parameter" to "context".
*Suggest to specify issue for each return code, where multiple failure return codes are listed, or generalize.
*Minor improvements to parameter documentation proposed by eng.

### CODE REVIEW

File: include/mbedtls/dhm.h [hunk 1]
```diff
@@ -1,7 +1,12 @@
 /**
  * \file dhm.h
  *
- * \brief Diffie-Hellman-Merkle key exchange.
+ * \brief This file contains DHM definitions and functions.
```
Comment (hanno-arm) [inline include/mbedtls/dhm.h]: Whatever wording we decide on in https://github.com/ARMmbed/mbedtls/pull/1513#discussion_r180687267, we should use an analogous one here.

File: include/mbedtls/dhm.h [hunk 2]
```diff
@@ -155,63 +153,70 @@ int mbedtls_dhm_read_params( mbedtls_dhm_context *ctx,
...
- *                 on failure.
+ * \return         \c 0 on success.
+ * \return         An \c MBEDTLS_ERR_DHM_XXX error code on failure.
  */
 int mbedtls_dhm_read_public( mbedtls_dhm_context *ctx,
                      const unsigned char *input, size_t ilen );
 
 /**
- * \brief          This function creates its own private value \c X and
+ * \brief          This function creates its own \c X private key and
```
Comment (hanno-arm) [inline include/mbedtls/dhm.h]: Same here: Why the reordering?

File: include/mbedtls/dhm.h [hunk 3]
```diff
@@ -261,8 +266,8 @@ void mbedtls_dhm_free( mbedtls_dhm_context *ctx );
  * \param dhminlen    The size of the buffer, including the terminating null
  *                    Byte for PEM data.
  *
- * \return            \c 0 on success, or a specific DHM or PEM error code
- *                    on failure.
+ * \return            \c 0 on success.
+ * \return            A specific DHM or PEM error code on failure.
```
Comment (hanno-arm) [inline include/mbedtls/dhm.h]: In line with previous wording, I'd prefer "An MBEDTLS_ERR_DHM_XXX or MBEDTLS_ERR_PEM_XXX error code on failure." here.

File: include/mbedtls/dhm.h (hunk 1 above)
Comment (RosZad) [reply include/mbedtls/dhm.h]: Is "This file provides an API for the DHM key exchange." accurate?

Comment (hanno-arm) [reply include/mbedtls/dhm.h]: I'd be fine with that. @gilles-peskine-arm ?

File: include/mbedtls/dhm.h (hunk 2 above)
Comment (RosZad) [reply include/mbedtls/dhm.h]: Correct English ordering...  either the new way, or if you really insist:
"This function creates its own private value, \c X, and"
The current phrasing is better...

Comment (hanno-arm) [reply include/mbedtls/dhm.h]: I prefer the second phrasing - let's have another opinion. @gilles-peskine-arm Ping.

Comment (gilles-peskine-arm) [reply include/mbedtls/dhm.h]: X is not an adjective here, it's a name, so “X private key” doesn't make sense to me. “His brother Bob”, “its private key X”.

Comment (hanno-arm) [reply include/mbedtls/dhm.h]: I'm used to "his brother Bob", too, but if that happens to be grammatically wrong (not sure about it), let's take "... private value, \c X, and ...", which should be comprehensible to anyone.

File: include/mbedtls/dhm.h (hunk 1 above)
Comment (RosZad) [reply include/mbedtls/dhm.h]: done

File: include/mbedtls/dhm.h [hunk 4]
```diff
@@ -155,63 +153,70 @@ int mbedtls_dhm_read_params( mbedtls_dhm_context *ctx,
...
+ * \return         \c 0 if successful.
+ * \return         An \c MBEDTLS_ERR_DHM_XXX error code on failure.
  */
 int mbedtls_dhm_set_group( mbedtls_dhm_context *ctx,
                            const mbedtls_mpi *P,
                            const mbedtls_mpi *G );
 
 /**
- * \brief          This function imports the public value G^Y of the peer.
+ * \brief          This function imports the G^Y public value of the peer.
```
[the remark this answers is not in the document]
Comment (RosZad) [reply include/mbedtls/dhm.h]: Changed to second phrasing

File: include/mbedtls/dhm.h (hunk 2 above)
Comment (RosZad) [reply include/mbedtls/dhm.h]: Changed to second phrasing

File: include/mbedtls/dhm.h (hunk 3 above)
Comment (RosZad) [reply include/mbedtls/dhm.h]: done

File: include/mbedtls/dhm.h [hunk 5]
```diff
@@ -275,8 +280,8 @@ int mbedtls_dhm_parse_dhm( mbedtls_dhm_context *dhm, const unsigned char *dhmin,
  * \param dhm      The DHM context to load the parameters to.
  * \param path     The filename to read the DHM parameters from.
  *
- * \return         \c 0 on success, or a specific DHM or PEM error code
- *                 on failure.
+ * \return         \c 0 on success.
+ * \return         A specific DHM or PEM error code on failure.
```
[the remark this answers is not in the document]
Comment (RosZad) [reply include/mbedtls/dhm.h]: done

File: include/mbedtls/dhm.h [hunk 6]
```diff
@@ -267,7 +268,8 @@ void mbedtls_dhm_free( mbedtls_dhm_context *ctx );
  *                    Byte for PEM data.
  *
  * \return            \c 0 on success.
- * \return            A specific DHM or PEM error code on failure.
+ * \return            An \c MBEDTLS_ERR_DHM_XXX or MBEDTLS_ERR_PEM_XXX error code
```
Comment (hanno-arm) [inline include/mbedtls/dhm.h]: Please add `\c` before `MBETLS_ERR_PEM_XXX`, too.

File: include/mbedtls/dhm.h [hunk 7]
```diff
@@ -281,7 +283,8 @@ int mbedtls_dhm_parse_dhm( mbedtls_dhm_context *dhm, const unsigned char *dhmin,
  * \param path     The filename to read the DHM parameters from.
  *
  * \return         \c 0 on success.
- * \return         A specific DHM or PEM error code on failure.
+ * \return            An \c MBEDTLS_ERR_DHM_XXX or MBEDTLS_ERR_PEM_XXX error code
```
Comment (hanno-arm) [inline include/mbedtls/dhm.h]: Same here.

File: include/mbedtls/dhm.h (hunk 6 above)
Comment (RosZad) [reply include/mbedtls/dhm.h]: done

File: include/mbedtls/dhm.h (hunk 7 above)
Comment (RosZad) [reply include/mbedtls/dhm.h]: done

File: include/mbedtls/dhm.h (hunk 2 above)
Comment (sbutcher-arm) [reply include/mbedtls/dhm.h]: Maybe I've misunderstood the comments above, but it doesn't look like the phrasing has been changed to the second option to me.

@RosZad - is the text as you intend?

**Comment by sbutcher-arm** [comment]
I'm not quite sure where a mistake was made, but I can see that the text in the PR now satisfies the review feedback, which answers my question above, so I can approve the PR, and mark it as 'ready for merge'.

**Pull request merged**